packages/runtime/container-runtime/src/test/summaryManager.spec.ts

```diff
@@ -0,0 +1,356 @@
+    }
+
+    let clientElection: TestSummarizerClientElection;
+    let connectedState: TestConnectedState;
+    let summarizer: TestSummarizer;
+    let requestCalls = 0;
+    let requestDeferred = new Deferred<void>();
+    const requestSummarizer = async (): Promise<ISummarizer> => {
+        summarizer = new TestSummarizer();
+        requestCalls++;
+        requestDeferred = new Deferred();
+        await requestDeferred.promise;
+        return summarizer;
+    };
+
+    function createSummaryManager({
+        connected = false,
+        ...config
+    }: Readonly<Partial<{ connected?: boolean; } & ISummaryManagerConfig>> = {}) {
+        connectedState = new TestConnectedState();
+        if (connected) {
+            connectedState.connect();
+        }
+        clientElection = new TestSummarizerClientElection();
+        summaryManager = new SummaryManager(
+            clientElection,
+            connectedState,
+            summaryCollection,
+            mockLogger,
+            requestSummarizer,
+            throttler,
+            config,
+        );
+        summaryManager.start();
+    }
+
+    function assertState(expectedState: SummaryManagerState, message: string) {
+        assert.strictEqual(summaryManager.currentState, expectedState, message);
+    }
+
+    function assertRequests(count: number, message?: string) {
+        const prefix = message ? `${message} - ` : "";
+        assert.strictEqual(requestCalls, count, `${prefix}Unexpected request count`);
+    }
+
+    afterEach(() => {
+        clientElection.removeAllListeners();
+        summarizer.removeAllListeners();
+        summaryManager.removeAllListeners();
+        connectedState.removeAllListeners();
+        throttler.delayMs = 0;
+        summaryCollection.opsSinceLastAck = 0;
+        requestCalls = 0;
+        clock.reset();
+    });
+
+    it("Should become summarizer if connected, then elected; stop summarizer after disconnect", async () => {
+        createSummaryManager({ opsToBypassInitialDelay: 0 });
+        assertState(SummaryManagerState.Off, "should start off");
+        connectedState.connect();
+        await flushPromises();
+        assertState(SummaryManagerState.Off, "connected but not yet elected");
+        clientElection.electClient("other");
+        await flushPromises();
+        assertState(SummaryManagerState.Off, "connected but other client elected");
+        clientElection.electClient(thisClientId);
+        await flushPromises();
+        assertState(SummaryManagerState.Starting, "should request summarizer");
+        assertRequests(1, "should have requested summarizer");
+        requestDeferred.resolve();
+        await flushPromises();
+        assertState(SummaryManagerState.Running, "summarizer should be running");
+        connectedState.disconnect();
+        await flushPromises();
+        assertState(SummaryManagerState.Stopping, "should be stopping after disconnect");
+        summarizer.runDeferred.resolve();
```

Comment:
Consider adding a comment why this is resolved here (it's not super obvious). Same in other places.

Reply:
Changed to a self-describing function, with a comment at the function definition.